src/app/profile/update/update-guard.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { ActivatedRoute, CanActivate, Router } from '@angular/router';
+
+import { Observable, Subscription } from 'rxjs';
+
+import { AuthenticationService, UserService, User } from 'ngx-login-client';
+
+import { LoginService } from '../../shared/login.service';
+
+@Injectable()
+export class UpdateGuard implements CanActivate, OnDestroy {
+  subscription: Subscription;
+  activateObservable: Observable<boolean>;
+  differentUsername = true;
+  username = '';
+
+  constructor(
+    private route: ActivatedRoute,
+    private authService: AuthenticationService,
+    private router: Router,
+    private loginService: LoginService,
+    private userService: UserService
+  ) {
+    this.activateObservable = Observable.create(observer => {
+      this.subscription = this.userService.loggedInUser.subscribe(user => {
+        this.handleUserNotification(user);
+        if (this.differentUsername) {
```

Comment:
You can simplify this by removing this.differentUsername and just returning a boolean from the handleUserNotification maybe renaming that function to indicate what it's actually checking.